### DISCUSSION TITLE
OSError

### DISCUSSION THREAD

**Discussion by Pookeyblow** [khoj-ai/khoj#168] [Q&A]
Hi!

I'm trying to install Khoj for Obsidian and I'm getting the following error in Terminal:

> ERROR: Could not install packages due to an OSError: Cannot move the non-empty directory '/opt/homebrew/lib/python3.10/site-packages/PIL/': Lacking write permission to '/opt/homebrew/lib/python3.10/site-packages/PIL/'.

Does anyone know how to fix this?

I'm on macOS Monterey 12.1 (M1 iMac).

**Comment by debanjum** [comment]
Based on the error log you've shared. The user you're trying to install khoj with doesn't have the same system privileges as the user you installed the Pillow Python Package (aka PIL) earlier as. For e.g, maybe you installed PIl with sudo? Somethings to try:

1. You can manually delete the PIL directory at `/opt/homebrew/lib/python3.10/site-packages/PIL/` and retry installing khoj to see if it resolves your issue
2. You can install khoj in a virtual environment. For e.g by using [pipx](https://pypa.github.io/pipx/#install-pipx) with `pip install pipx && pipx install khoj-assistant`. This isolates the environment in which khoj and its dependencies are installed and help avoid the permission issue you're facing

Let me know if either of those suggestions fix your issue?
